test/base.js

```diff
@@ -701,6 +701,18 @@ describe('Base', function () {
 
       assert.equal(gen.options.undef, undefined);
     });
+
+    it('disallows Boolean options starting with no-', function () {
+      var generator = new this.Dummy([], {
+        env: this.env,
+        resolved: 'test'
+      });
+      var addWrongOp = function() {
+        generator.option('no-op');
+      };
+      assert.throws(addWrongOp, Error, 'Does not throw Error');
```

Comment:
Don't add a third argument to assertions. These usually make the error less clear than the default messages.